### DISCUSSION TITLE
SMBIOS 3.x support in iPXE anytime soon?

### DISCUSSION THREAD

**Discussion by ajyoung-oracle** [ipxe/ipxe#180] [General]
It appears that iPXE only currently supports SMBIOS 2.x tables at present (i.e. include/ipxe/smbios.h has only 2.x based struct defs, SMBIOS_SIGNATURE is 2.x, etc.).

Are there any plans to support (or anyone working on) SMBIOS 3.x support?

We have noticed that on Arm64 machines, the SMBIOS structures passed from qemu->OVMF are 3.x based and thus iPXE cannot parse the SMBIOS tables resulting in missing settings.

 Thanks in advance for any help/input/comments/suggestions/etc,

 -Aaron

**Comment by mcb30** [comment]
Nothing planned.  Based on a very quick read of the SMBIOS spec, it looks as though the only real difference is in the format of the table that identifies the address in memory the SMBIOS structures are to be found.  iPXE parses this table exactly once and thereafter retains only the SMBIOS structures address.

**Reply by mcb30** [reply]
This quick read seems to be correct.  Raised implementation as issue #192 and implemented in commit https://github.com/ipxe/ipxe/commit/dced22d6d.  Should now work as expected.

**Comment by ajyoung-oracle** [comment]
Confirmed that commit fixes the issue. iPXE successfully obtains proper settings from SMBIOS 3.x tables.

Thanks for fixing this.

NOTE: I actually had a patch implemented to fix this, which I had planned to submit as a PR. It was nearly identical to your commit. Only significant difference was I treated new smbios3_entry struct signature as an array of 5 bytes (instead of a uint32_t) and then just checked signature from efi_find_smbios() using strncmp() - which may be a bit better since it allows to check the full 5 byte sig instead of just the first 4 bytes.
 
Thanks!
